Nef_3/include/CGAL/Nef_3/SNC_point_locator.h

```diff
@@ -250,22 +257,25 @@ class SNC_point_locator_by_spatial_subdivision :
         else if( CGAL::assign( e, *o) && ((mask&2) != 0)) {
           Point_3 q;
           _CGAL_NEF_TRACEN("trying edge on "<< Segment_3(e->source()->point(),e->twin()->source()->point()));
-          if( is.does_intersect_internally( ray, Segment_3(e->source()->point(),
-                                                           e->twin()->source()->point()), q)) {
-            _CGAL_NEF_TRACEN("ray intersects edge on "<<q);
-            _CGAL_NEF_TRACEN("prev. intersection? "<<hit);
-            CGAL_assertion_code
-              (if( hit) _CGAL_NEF_TRACEN("prev. intersection on "<<eor));
-            if( hit && !has_smaller_distance_to_point( ray.source(), q, eor))
-              continue;
-            _CGAL_NEF_TRACEN("is the intersection point on the current cell? "<<
-                    candidate_provider->is_point_on_cell( q, objects_iterator));
-            if( !candidate_provider->is_point_on_cell( q, objects_iterator))
-                continue;
-            eor = q;
-            result = make_object(e);
-            hit = true;
-            _CGAL_NEF_TRACEN("the edge becomes the new hit object");
+          if ((ray_source_vertex != Vertex_handle()) && (ray_source_vertex != e->source()) && (ray_source_vertex != e->twin()->source())) {
```

Comment:
You check for a null_handle here but not above?

Reply:
Yep, that's it.

In addition, I was wondering why it does it like this on line 260 but it is not the same on line 244.

Reply:
> In addition, I was wondering why it does it like this on line 260 but it is not the same on line 244.


Oh you've reverted commit [211cb75](https://github.com/CGAL/cgal/pull/6431/commits/211cb75dd34cad6a07d3ea2ed18446c5fb36879a) Well that explains the second question at least ;)